rust/parquet/src/schema/types.rs

```diff
@@ -479,13 +572,17 @@ impl<'a> GroupTypeBuilder<'a> {
 
     /// Creates a new `GroupType` instance from the gathered attributes.
     pub fn build(self) -> Result<Type> {
-        let basic_info = BasicTypeInfo {
+        let mut basic_info = BasicTypeInfo {
             name: String::from(self.name),
             repetition: self.repetition,
             converted_type: self.converted_type,
-            logical_type: self.logical_type,
+            logical_type: self.logical_type.clone(),
             id: self.id,
         };
+        // Populate the converted type if only the logical type is populated
```

Comment:
we might need more tests for the case when logical type is set.

Reply:
I added a test for the group type (modified an existing one), but for primitive types, I need the schema printer + parser. So, I'll increase the test coverage as part of #9705